packages/lightcone_v2.js/src/sign/exchange.ts

```diff
@@ -99,42 +98,52 @@ export class Exchange {
 
     public flattenList = (l: any[]) => {
         return [].concat.apply([], l);
-    }
+    };
 
+    // JavaScript: Promises and Why Async/Await Wins the Battle
+    // https://hackernoon.com/javascript-promises-and-why-async-await-wins-the-battle-4fc9d15d509f
+    // ES7 async error handling: do it right or die
+    // https://medium.com/@giovannipinto/async-error-handling-forced-to-do-it-right-2817cf9e8b43
     public async createAccount(wallet: WalletAccount, gasPrice: number) {
-        // TODO: need to check if gasPrice is a reasonable value
-        if (this.accounts.get(wallet) == null) {
-            const keyPair = generateKeyPair();
-            console.log(keyPair)
-            this.createOrUpdateAccount(keyPair.publicKeyX, keyPair.publicKeyY, gasPrice).then((rawTx: Transaction) => {
-                    const signedTx = wallet.signEthereumTx(rawTx);
-                    wallet.sendTransaction(new Eth(''), signedTx).then(() => { // TODO: config
-                        grpcClientService.getAccount(wallet.getAddress()).then((account: Account) => {
-                            const dexAccount = new DexAccount();
-                            dexAccount.nonce = 0;
-                            dexAccount.owner = wallet.getAddress();
-                            dexAccount.accountID = account.getAccountId().getValue();
-                            dexAccount.publicKeyX = keyPair.publicKeyX;
-                            dexAccount.publicKeyY = keyPair.publicKeyY;
-                            dexAccount.secretKey = keyPair.secretKey;
-                            this.accounts.set(wallet, dexAccount);
-                            this.currentDexAccount = dexAccount;
-                            this.currentWalletAccount = wallet;
-                        });
+        try {
+            // TODO: need to check if gasPrice is a reasonable value
+            if (this.accounts.get(wallet) == null) {
+                const keyPair = generateKeyPair();
+                this.currentWalletAccount = wallet;
+                let rawTx: Transaction = await this.createOrUpdateAccount(keyPair.publicKeyX, keyPair.publicKeyY, gasPrice)
+                const signedTx = wallet.signEthereumTx(rawTx);
+
+                // TODO: Let's avoid using Promises. Change this part to await.
+                // At least, avoid using a Promise in another Promise.
```

Comment:
@iwbrhwfy Please see https://hackernoon.com/javascript-promises-and-why-async-await-wins-the-battle-4fc9d15d509f